# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: devtools/shared/specs/targets/worker.js
# Commit: 6eaa4d12b58c
# Full Hash: 6eaa4d12b58cb4762729fdf53511450926383db8
# Author: Nicolas Chevobbe <nchevobbe@mozilla.com>
# Date: 2020-10-15 09:40:06
# Regressor Bug: 1633712
# File Overlap Count: 1
# Description:
#   Bug 1633712 - [devtools] Create WorkerTargets as soon as possible. r=ochameau,devtools-backward-compat-reviewers.
#   
#   This patch adds support for dedicated worker targets in the Watcher actor.
#   Shared and Service workers are not handled yet.
#   
# ==============================================================================

diff -r 10f97f47c883 -r 6eaa4d12b58c devtools/shared/specs/targets/worker.js
--- a/devtools/shared/specs/targets/worker.js	Thu Oct 15 05:11:58 2020 +0000
+++ b/devtools/shared/specs/targets/worker.js	Thu Oct 15 05:17:52 2020 +0000
@@ -3,15 +3,17 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 "use strict";
 
-const { generateActorSpec } = require("devtools/shared/protocol");
-
-// XXX: This actor doesn't expose any methods and events yet, but will in the future
-// (e.g. resource-available-form and resource-destroyed-form).
+const { Arg, generateActorSpec } = require("devtools/shared/protocol");
 
 const workerTargetSpec = generateActorSpec({
   typeName: "workerTarget",
   methods: {},
-  events: {},
+  events: {
+    "resource-available-form": {
+      type: "resource-available-form",
+      resources: Arg(0, "array:json"),
+    },
+  },
 });
 
 exports.workerTargetSpec = workerTargetSpec;
